integrationTests/state/stateTrie/stateTrie_test.go

```diff
@@ -2166,3 +2136,238 @@ func TestProofAndVerifyProofDataTrie(t *testing.T) {
 		assert.Equal(t, value, dataTrieProof.Value)
 	}
 }
+
+func TestTrieDBPruning_PruningOldData(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+	adb := createAccountsDBTestSetup()
+	numAccounts := uint32(10000)
+	numIncreaseDecreaseIterations := 100
+
+	rootHashes := make([][]byte, 0)
+	rootHash, err := createDummyAccountsWith100EGLD(numAccounts, adb)
+	require.Nil(t, err)
+	rootHashes = append(rootHashes, rootHash)
+
+	for i := 0; i < numIncreaseDecreaseIterations; i++ {
+		// change some accounts
+		rootHash, err = increaseBalanceForAccountsStartingWithIndex(100, 1000, 10, adb)
+		require.Nil(t, err)
+		rootHashes = append(rootHashes, rootHash)
+		checkAccountsBalances(t, 100, 1000, 110, adb)
+
+		// change same accounts state back
+		rootHash, err = decreaseBalanceForAccountsStartingWithIndex(100, 1000, 10, adb)
+		require.Nil(t, err)
+		rootHashes = append(rootHashes, rootHash)
+		adb.PruneTrie(rootHashes[len(rootHashes)-2], state.OldRoot)
+		checkAccountsBalances(t, 100, 1000, 100, adb)
+	}
+}
+
+func TestTrieDBPruning_PruningOldDataWithDataTries(t *testing.T) {
+	if testing.Short() {
```

Comment:
Is this a long test? Maybe we should skip it if it takes too long? Or adjust the parameters?